src/vs/workbench/services/search/common/searchExtTypesInternal.ts

```diff
@@ -0,0 +1,13 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import type { FileSearchOptions, TextSearchOptions } from './searchExtTypes';
+
+interface SearchOptionsExtendedCommon {
+	numThreads?: number | undefined;
+}
+
+export interface TextSearchOptionsExtended extends TextSearchOptions, SearchOptionsExtendedCommon { }
```

Comment:
These are specific to ripgrep, so perhaps something like `RipgrepTextSearchOptions` would make this more specific.